Sender/testBMSData.py

```diff
@@ -0,0 +1,13 @@
+import unittest
+import GenerateBMSData
+class test_streaming_bms_data(unittest.TestCase):
+    def test_format_battery_parameter(self):
+        bms_param = GenerateBMSData.generateBatteryParameters()
```

Comment:
This test is good, it says that the output will contain two parameters. However, it does not say whether those parameters will be in json, whether they will be string or number.
Adding a test to parse the output will ensure that the format is documented, and also that future developers will know if they are violating.
E.g., your receiver has assumed that the temperature would come first and soc will come next. If any maintainer changes this in future, they will know only when they integrate with the updated receiver. A test for this would have made the whole process faster ('fail fast').